app/lib/common/providers/room_providers.dart

```diff
@@ -198,6 +199,70 @@ final canonicalParentProvider = FutureProvider.autoDispose
   }
 });
 
+/// Get all the parents of the space. Errors if no relation is found.Stays up
+/// to date with underlying client data if a space was found.
+final parentsProvider = FutureProvider.autoDispose
+    .family<List<SpaceWithProfileData>?, String>((ref, roomId) async {
+  try {
+    List<SpaceWithProfileData> parentList = List.empty(growable: true);
+    final relations = await ref.watch(spaceRelationsProvider(roomId).future);
+    if (relations == null) {
+      return null;
+    }
+
+    // Collect all parents: mainParent and otherParents
+    final allParents = [];
+    final mainParent = relations.mainParent();
+    if (mainParent != null) {
+      allParents.add(mainParent);
+    }
+    allParents.addAll(relations.otherParents());
+
+    for (final parent in allParents) {
+      final parentSpace = await ref
+          .watch(maybeSpaceProvider(parent.roomId().toString()).future);
+      if (parentSpace == null) {
+        // not found, skipping it
+        continue;
+      }
+      final profile =
+          await ref.watch(spaceProfileDataProvider(parentSpace).future);
+      final SpaceWithProfileData data = (space: parentSpace, profile: profile);
+      parentList.add(data);
```

Comment:
this logic is still flawed for the same reason I brought up in the last review...